pkg/extension/sumologicextension/extension.go

```diff
@@ -592,6 +602,157 @@ func (se *SumologicExtension) sendHeartbeatWithHTTPClient(ctx context.Context, h
 	return nil
 }
 
+func getHostIpAddress() (string, error) {
+	c, err := net.Dial("udp", "8.8.8.8:53")
+	if err != nil {
+		return "", err
+	}
+
+	defer c.Close()
+	a := c.LocalAddr().(*net.UDPAddr)
+	return a.String(), nil
+}
+
+func getProxyInfo() (string, int, error) {
+	p := os.Getenv("HTTPS_PROXY")
+	if p == "" {
+		p = os.Getenv("HTTP_PROXY")
+	}
+
+	if p == "" {
+		return "", 0, nil
+	}
+
+	u, err := url.Parse(p)
+	if err != nil {
+		return "", 0, err
+	}
+
+	h, ps, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return "", 0, err
+	}
+
+	pi, err := strconv.Atoi(ps)
+	if err != nil {
+		return "", 0, err
+	}
+
+	return h, pi, nil
+}
+
+func (se *SumologicExtension) updateMetadataWithHTTPClient(ctx context.Context, httpClient *http.Client) error {
+	u, err := url.Parse(se.BaseUrl() + metadataUrl)
+	if err != nil {
+		return fmt.Errorf("unable to parse metadata URL %w", err)
+	}
+
+	info, err := host.Info()
+	if err != nil {
+		return err
+	}
+
+	ip, err := getHostIpAddress()
+	if err != nil {
+		return err
+	}
+
+	pAddr, pPort, err := getProxyInfo()
+	if err != nil {
+		return err
+	}
+
+	var buff bytes.Buffer
+	if err = json.NewEncoder(&buff).Encode(api.OpenMetadataRequestPayload{
+		HostDetails: api.OpenMetadataHostDetails{
+			Name:        info.Hostname,
+			OsName:      info.OS,
+			OsVersion:   info.PlatformVersion,
+			Environment: se.conf.CollectorDescription, // TODO: Replace with new configuration attribute?
+		},
+		AgentDetails: api.OpenMetadataAgentDetails{
+			RunningVersion: "1.0.0",
```

Comment:
@astencel-sumo How can I determine the agent running version?

Reply:
@sumo-drosiek any ideas?

Reply:
in `createExtension` you have access to `params.BuildInfo.Version`.

working example:
```
func createExtension(_ context.Context, params extension.CreateSettings, cfg component.Config) (extension.Extension, error) {
	fmt.Printf("current version: %s\n", params.BuildInfo.Version)
	config := cfg.(*Config)
	return newSumologicExtension(config, params.Logger, params.ID)
}
```